dali/python/CMakeLists.txt

```
@@ -43,3 +43,33 @@ copy_post_build(dali_python "${PROJECT_SOURCE_DIR}/Acknowledgements.txt" "${PROJ
 copy_post_build(dali_python "${PROJECT_SOURCE_DIR}/COPYRIGHT" "${PROJECT_BINARY_DIR}/dali/python/nvidia/dali")
 copy_post_build(dali_python "${PROJECT_SOURCE_DIR}/LICENSE" "${PROJECT_BINARY_DIR}/dali/python/nvidia/dali")
 copy_post_build(dali_python "${PROJECT_SOURCE_DIR}/tools/find_dali.cmake" "${PROJECT_BINARY_DIR}/dali/cmake")
+
+if(NOT ${CMAKE_CROSSCOMPILING})
+  # Prepate PYTHONPATH targetting the pre-packed wheel produced in this build
+  set(PYTHON_TARGET_PATH ${PROJECT_BINARY_DIR}/dali/python)
+  if($ENV{PYTHONPATH})
+    set(PYTHONPATH "${PYTHON_TARGET_PATH}:$ENV{PYTHONPATH}")
+  else()
+    set(PYTHONPATH "${PYTHON_TARGET_PATH}")
+  endif()
+
+  add_custom_target(dali_python_generate_stubs ALL DEPENDS dali_python)
+
+  # Build the .pyi stubs, adjusting the PYTHONPATH for the invokation, allowing to use the
+  # backend from the current build.
+  add_custom_command(
+    TARGET dali_python_generate_stubs
+    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+    COMMAND PYTHONPATH=${PYTHONPATH} ${PROJECT_SOURCE_DIR}/tools/python_stub_generator.py --wheel_path ${PROJECT_BINARY_DIR}/dali/python/nvidia/dali
+  )
+
+  # Format the bindings, so they are somewhat readable.
+  if (${FORMAT_PYTHON_STUB_FILES})
+    add_custom_target(dali_python_format_stubs ALL DEPENDS dali_python dali_python_generate_stubs)
+    add_custom_command(
+      TARGET dali_python_format_stubs
+      WORKING_DIRECTORY ${PROJECT_BINARY_DIR}
+      COMMAND find dali/python/nvidia/dali -name '*.pyi' -print0 | xargs -0 yapf --in-place --verbose --parallel --style "${PROJECT_SOURCE_DIR}/.style.yapf"
```

This takes _ages_. I recommend using `black` instead.
```suggestion
      COMMAND COMMAND find dali/python/nvidia/dali -name '*.pyi' | xargs black -l 120 -t py38
```

---

If we agreed to use `black` uniformly, than yes, but we already have formatting rules for `yapf`.

---

These responses were posted deliberately as means of psychological torture on a person who holds beliefs of a minority. Shall I report it? I'm being harassed.